pyiron/atomistics/structure/atoms.py

```diff
@@ -1227,30 +1227,102 @@ def _get_flattened_orientation(view_plane, distance_from_camera):
+                third component (if specified) is the vertical component, which is ignored and calculated internally.
+                The orthonormality of the orientation is internally ensured, and therefore is not required in the
+                function call.
+            distance_from_camera (float): Distance of the camera from the structure. Higher = farther away.
+
+        Returns:
+            (list): Flattened list of len = 16, which is the input argument to `view.contol.orient`
+        """
+        if distance_from_camera <= 0:
+            raise ValueError("´distance_from_camera´ must be a positive float!")
         flattened_orientation = np.eye(4)
-        flattened_orientation[:3, :3] = rotation_matrix
+        flattened_orientation[:3, :3] = self._get_orientation(view_plane)
 
         return (distance_from_camera * flattened_orientation).ravel().tolist()
 
+    def plot3d_plotly(
+        self,
+        scalar_field=None,
+        particle_size=1.0,
+        camera="orthographic",
+        view_plane=np.array([1, 1, 1]),
+        distance_from_camera=1.25,
+    ):
+        """
+
+        Args:
+            camera (str): 'perspective' or 'orthographic'. (Default is 'perspective'.)
+            particle_size (float): Size of the particles. (Default is 1.)
+            scalar_field (numpy.ndarray): Color each atom according to the array value (Default is None, use coloring
+                scheme.)
+            view_plane (numpy.ndarray): A Nx3-array (N = 1,2,3); the first 3d-component of the array specifies
+                which plane of the system to view (for example, [1, 0, 0], [1, 1, 0] or the [1, 1, 1] planes), the
+                second 3d-component (if specified, otherwise [1, 0, 0]) gives the horizontal direction, and the third
+                component (if specified) is the vertical component, which is ignored and calculated internally. The
+                orthonormality of the orientation is internally ensured, and therefore is not required in the function
+                call. (Default is np.array([0, 0, 1]), which is view normal to the x-y plane.)
+            distance_from_camera (float): Distance of the camera from the structure. Higher = farther away.
+                (Default is 14, which also seems to be the NGLView default value.)
+
+        Returns:
+            (plotly.express): The NGLView widget itself, which can be operated on further or viewed as-is.
+
+        """
+        try:
+            import plotly.express as px
+        except ImportError:
+            print("plotly not installed")
+            return
+        parent_basis = self.get_parent_basis()
+        elements = parent_basis.get_chemical_symbols()
+        atomic_numbers = parent_basis.get_atomic_numbers()
+        positions = self.positions
+        if scalar_field is None:
+            scalar_field = elements
+        fig = px.scatter_3d(x=positions[:,0],
```

Comment:
If x, y, and z are the first three function arguments, you should be able to instead use

```python
fig = px.scatter_3d(
    *self.positions.T,
    color=scalar_field,
    opacity=1,
    size=that_long_thing
)
```

Personally I like this better, but if I'm in the minority I don't particularly object to how it is now. This format goes well with my earlier comment about not defining a `positions` variable though, because then we're only calling `self.positions` once anyhow.

Reply:
If it's for example inside a notebook, I agree. But if it's inside pyiron, I prefer to explicitly write which argument was meant, because they might change the order on their side.

Reply:
Ah and they are anyway not the first arguments

Reply:
I haven't tried it out, but I wonder if comma assignment works inside functions, eg f(x,y,z=*vector.T). Just on my phone now, but I'll experiment later. Sadly, I suspect it doesn't.

Reply:
I tried, but it unfortunately didn't work